attach/nv_attach_impl/nv_attach_impl.cpp

```diff
@@ -54,6 +56,56 @@
 #include "ptx_pass_config.h"
 using namespace bpftime;
 using namespace attach;
+static nv_attach_impl *g_nv_attach_impl_instance = nullptr;
+namespace {
+template <typename FnPtr>
+FnPtr resolve_symbol(FnPtr &cached, const char *name)
+{
+	if (!cached) {
+		cached = reinterpret_cast<FnPtr>(dlsym(RTLD_NEXT, name));
+		if (!cached) {
+			SPDLOG_ERROR("Failed to resolve symbol {}", name);
+		}
+	}
+	return cached;
+}
+using nvrtc_get_ptx_size_fn = nvrtcResult (*)(nvrtcProgram, size_t *);
+using nvrtc_get_ptx_fn = nvrtcResult (*)(nvrtcProgram, char *);
+using nvrtc_destroy_program_fn = nvrtcResult (*)(nvrtcProgram *);
+static nvrtc_get_ptx_size_fn real_nvrtc_get_ptx_size = nullptr;
+static nvrtc_get_ptx_fn real_nvrtc_get_ptx = nullptr;
+static nvrtc_destroy_program_fn real_nvrtc_destroy_program = nullptr;
+using cu_module_load_data_fn =
+	CUresult (*)(CUmodule *, const void *);
+using cu_module_load_data_ex_fn = CUresult (*)(CUmodule *, const void *,
+					       unsigned int, CUjit_option *,
+					       void **);
+static cu_module_load_data_fn real_cu_module_load_data = nullptr;
+static cu_module_load_data_ex_fn real_cu_module_load_data_ex = nullptr;
+
+static nvrtc_get_ptx_size_fn get_real_nvrtc_get_ptx_size()
+{
```

Comment:
is it better and more clear to put these resolve_symbol into a new header?  and give it a better name like resolve_xxx_function_symbol